mejuri-inc/flowcommerce_spree
# [TEC-4657] Adding basic rspec configuration along with some specs for currentZoneLoader

### What problem is the code solving?
Adding basic configuration to run specs.

### How does this change address the problem?
- Configures postgresql instead of sqlite3, in order to match the schema we are working with mejuri-web. Unfortunately using squlite3 is not 100% compatible with the jsonb querying pattern we are using in mejuri-web (for example: `where("meta -> 'flow_data' ->> 'country' = ?", ... )` does not work.
- Adds basic rspec configuration files.
- Configures factorybot.
- Updating spec/dummy application with some shared code from mejuri-web (DB schema, models and controllers). 

### Why is this the best solution?
We need specs coverage. This is a first step to achieve a good coverage. 

### Share the knowledge
**Steps to run test**

1. Make sure that `spec/dummy/db/schema.rb` file is up to date. This is to make sure we are testing with latest relevant changes performed in the Database that might affect Flow integration.
2. Update the `spec/dummy/config/database.yml` with correct credentials for the postgresql DB. (if you are using docker to load mejuri-web, you can use the same credentials as your mejuri-web app).
3. Go to spec/dummy and run the following commands:
```
rake db:create RAILS_ENV=test
rake db:schema:load RAILS_ENV=test
rake db:migrate RAILS_ENV=test
```
4. From the application's root directory you can run specs now.
